src/builtins.js

```diff
@@ -112,7 +118,7 @@ class AjaxArrow extends SimpleAsyncArrow {
             var c = s.substring(i + 2, j);
 
             var ncs = new ConstraintSet([]);
-            var err = [new NamedType('AjaxError')];
```

Comment:
This is actually the type of error that can occur from an ajax arrow. This isn't a composition-time error.

Can you make it so that AjaxArrow and DBArrow can supply their own error types for this constructor (AjaxError and QueryError for each subclass)?